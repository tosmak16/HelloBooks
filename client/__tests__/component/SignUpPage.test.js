import React from 'react';
import { mount } from 'enzyme';
import render from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import expect from 'expect';


import ConnectedSignUpPage, { SignUpPage } from '../../src/components/SignUpPage';

const mockFuction = jest.fn();

describe('SignupPage', () => {
  const initialState = {
    register: [{
      isRegistered: false,
      data: {},
      error: '',
      isFetching: false,
      response: {}
    }]
  };
  const mockStore = configureMockStore();
  let store;
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}><ConnectedSignUpPage /></Provider>);
  });


  it('should be defined', () => {
    expect(SignUpPage).toBeDefined();
  });

  it('should test and take snapshot of SignUpPage', () => {
    const tree = render.create(<SignUpPage signup={mockFuction} register={{ error: '' }} />);

    expect(tree).toMatchSnapshot();
  });
  it('it should render the ConnectedSignUpPage component', () => {
    expect(wrapper.find(ConnectedSignUpPage).length).toEqual(1);
  });

  it('it should check Prop matches with initialState', () => {
    expect(wrapper.find(SignUpPage).prop('register')).toEqual(initialState.register[0]);
  });
});
