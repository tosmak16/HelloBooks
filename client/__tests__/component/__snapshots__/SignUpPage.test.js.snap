// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupPage should test and take snapshot of SignUpPage 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col s12 col l12 col m12"
    >
      <div
        className="col s12 col m6 offset-m4 col 16 offset-l4"
        id="signup_border"
      >
        <div
          className="row"
          id="signUpForm"
        >
          <form
            action="/login"
            className="form-signin"
            onSubmit={[Function]}
          >
            <h4
              className="form-signin-heading"
            >
              Create an account
            </h4>
            
            <div
              className="row"
            >
              <div
                className="col input-field s12"
              >
                <input
                  className="form-control validate"
                  defaultValue={undefined}
                  id="firstname"
                  name="firstName"
                  onChange={[Function]}
                  placeholder={undefined}
                  required={true}
                  type="text"
                  value=""
                />
                <label
                  className=""
                  data-error={undefined}
                  data-success={undefined}
                  htmlFor="firstname"
                >
                  First Name
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col input-field s12"
              >
                <input
                  className="form-control validate"
                  defaultValue={undefined}
                  id="lastname"
                  name="lastName"
                  onChange={[Function]}
                  placeholder={undefined}
                  required={true}
                  type="text"
                  value=""
                />
                <label
                  className=""
                  data-error={undefined}
                  data-success={undefined}
                  htmlFor="lastname"
                >
                  Last Name
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col input-field s12"
              >
                <input
                  className="form-control validate"
                  defaultValue={undefined}
                  id="inputSignUpEmail"
                  name="email"
                  onChange={[Function]}
                  placeholder={undefined}
                  required={true}
                  type="email"
                  value=""
                />
                <label
                  className=""
                  data-error={undefined}
                  data-success={undefined}
                  htmlFor="inputSignUpEmail"
                >
                  Email address
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col input-field s12"
              >
                <input
                  className="form-control validate"
                  defaultValue={undefined}
                  id="inputUsername"
                  name="username"
                  onChange={[Function]}
                  placeholder={undefined}
                  required={true}
                  type="text"
                  value=""
                />
                <label
                  className=""
                  data-error={undefined}
                  data-success={undefined}
                  htmlFor="inputUsername"
                >
                  Username
                </label>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col input-field s12"
              >
                <input
                  className="form-control validate"
                  defaultValue={undefined}
                  id="inputsSignUpPassword"
                  name="password"
                  onChange={[Function]}
                  placeholder={undefined}
                  required={true}
                  type="password"
                  value=""
                />
                <label
                  className=""
                  data-error={undefined}
                  data-success={undefined}
                  htmlFor="inputsSignUpPassword"
                >
                  Password
                </label>
              </div>
            </div>
            <a
              href=""
            >
              <button
                className="btn waves-effect waves-light btn btn-lg btn-success btn-block"
                disabled={false}
                id="signUbtn"
                onClick={undefined}
                type="submit"
              >
                Sign up
              </button>
            </a>
          </form>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="footer-distributed "
  >
    <div
      className="row"
    >
      <div
        className=" col l12 col m12 col s12"
      >
        <div
          className="footer-left col l4 col m4 col s12"
        >
          <h3>
            Hello
            <span>
              Books
            </span>
          </h3>
          <p
            className="footer-links"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Blog
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Pricing
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              About
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Faq
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              Contact
            </a>
          </p>
          <p
            className="footer-company-name"
          >
            Tosmak © 2017
          </p>
        </div>
        <div
          className="footer-center col l4 col m4 col s12"
        >
          <div>
            <i
              className="fa fa-map-marker"
            />
            <p>
              <span>
                21 Ori-oke Street
              </span>
               Lagos, Nigeria
            </p>
          </div>
          <div>
            <i
              className="fa fa-phone"
            />
            <p>
              08127112053
            </p>
          </div>
          <div>
            <i
              className="fa fa-envelope"
            />
            <p>
              <a
                href="tosmak16@gmail.com"
                onClick={[Function]}
                style={Object {}}
              >
                tosmak16@gmail.com
              </a>
            </p>
          </div>
        </div>
        <div
          className="footer-right col l4 col m14 col s12"
        >
          <div
            className="footer-company-about"
          >
            <span>
              About Hello Book
            </span>
            <p>
              it is a simple application that helps manage a library and its processes like stocking,tracking and renting books
            </p>
          </div>
          <div
            className="footer-icons"
          >
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="fa fa-facebook"
              />
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="fa fa-twitter"
              />
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="fa fa-linkedin"
              />
            </a>
            <a
              onClick={[Function]}
              style={Object {}}
            >
              <i
                className="fa fa-github"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
